File: 수학(구현)/백준_2004_조합 0의 개수/조합 0의 개수.java
import java.util.*;

public class Main {
    static int count_zero(int n, int k) {
        int cnt = 0;
        for (long i = k; i <= n; i *= k) {
            cnt += (n / i);
        }
        return cnt;
    }

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);

        int n = s.nextInt();
        int m = s.nextInt();

        int f = count_zero(n, 5) - count_zero(m, 5) - count_zero(n - m, 5);
        int t = count_zero(n, 2) - count_zero(m, 2) - count_zero(n - m, 2);

        System.out.println(Math.min(f, t));
    }
}

File: 수학(구현)/조합/백준_2004_조합 0의 개수/조합 0의 개수.kt
import java.util.*

fun count_zero(n: Int, k: Int): Int {
    var cnt = 0
    var i: Long = k.toLong()
    while (i <= n) {
        cnt += (n / i).toInt()
        i *= k
    }
    return cnt
}

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    val n = s.nextInt()
    val m = s.nextInt()

    var f = count_zero(n, 5) - count_zero(m, 5) - count_zero(n - m, 5)
    var t = count_zero(n, 2) - count_zero(m, 2) - count_zero(n - m, 2)

    print(Math.min(f, t))
}
